### ISSUE TITLE
Refactor `conda clean` into a plugin

### ISSUE THREAD

**Issue by kenodegard** [conda/conda#14253]
### Checklist

- [X] I added a descriptive title
- [X] I searched open requests and couldn't find a duplicate

### What is the idea?

Refactor `conda clean` into a `conda_subcommands` plugin and add a new plugin hook for registering new cleanup tasks (e.g., `conda_cleanup_tasks`).

### Why is this needed?

`conda clean` is a great way to remove old or unused files from your `conda` installation. Currently it only finds and removes package caches (both unpackaged and bundled (`.tar.bz2`, `.tar.bz2.part`, `.conda`, and `.conda.part`) packages), index caches, temporary files (`.c~` and `.trash` files), and logfiles.

With the plugin architecture other plugins are now able to interface more closely with conda and they bring their own special files into the picture. E.g., conda-content-trust has its own folder where it collects signatures, conda-libmamba-solver creates cached solve files, etc.

Providing a standard framework for allowing these other plugins to extend the functionality of `conda clean` much the same way we do for `conda doctor` will go a long way.

### What should happen?

1. introduce a new `conda_cleanup_tasks` plugin hook
2. refactor `conda.cli.main_clean` into `conda.plugins.subcommands.clean`

**Comment by jaimergp** [comment]
I get (and like!) (1) but what's the benefit of (2) 🤔? Curiosity, not trying to challenge the idea.

**Comment by kenodegard** [comment]
I agree (2) is not necessary. This started off with me mirroring what was done for `doctor`, now I'd argue that any subcommands that don't do anything on their own (as `conda doctor` and `conda clean` does here) should be refactored into `conda.plugins.subcommands`. If we refactor `conda info` in much the same way, it too would fit nicely into this pattern.

**Comment by jezdez** [comment]
Not stale

**Comment by danyeaw** [comment]
Another use case for this would be to remove the conda-pypi temporary channel when running `conda clean`.
